pkg/cache/v3/linear.go

```diff
@@ -185,6 +192,22 @@ func WithLogger(log log.Logger) LinearCacheOption {
 	}
 }
 
+// By default if a wildcard request is made to a go-control-plane
+// server backed by a linear cache, all the resources in the linear cache
+// will be returned.
+//
+// This scales poorly.
+//
+// If WithCustomWildcardMode is enabled, users will have to explicitly set for a given node
+// what wildcard resources are returned, instead or the linear cache returning every resource.
+// Users can set the wildcard resources for a given node with the method
+// LinearCache.UpdateWilcardResourcesForNode
+func WithCustomWildCardMode(mode bool) LinearCacheOption {
```

Comment:
Imo this should not take a parameter (refer to all the other boolean options on the cache)